plugins/catalog-backend/src/database/Database.ts

```diff
@@ -60,6 +60,11 @@ export class Database {
   }
 
   async addLocation(location: AddDatabaseLocation): Promise<DatabaseLocation> {
+    const existingLocation = await this.locationByTarget(location.target);
+    if (existingLocation) {
+      return existingLocation;
+    }
```

Comment:
you should do this in a transaction - all or nothing - see the code for `addOrUpdateComponent`

Reply:
for connection saving purpose? Cause otherwise we are inserting only once